labelbox/schema/project.py

```diff
@@ -160,6 +160,35 @@ def labels(self, datasets=None, order_by=None):
         return PaginatedCollection(self.client, query_str, {id_param: self.uid},
                                    ["project", "labels"], Label)
 
+    def export_queued_data_rows(self, timeout_seconds=120):
+        """ Returns all data rows that are currently enqueued for this project.
+
+        Args:
+            timeout_seconds (float): Max waiting time, in seconds.
+        Returns:
+            URL of the data file with this DataRow information. If the server didn't
+            generate during the `timeout_seconds` period, None is returned.
+        """
+        id_param = "projectId"
+        query_str = """mutation GetQueuedDataRowsExportUrlPyApi($%s: ID!)
+            {exportQueuedDataRows(data:{projectId: $%s }) {downloadUrl createdAt status} }
+        """ % (id_param, id_param)
+        sleep_time = 2
+        while True:
+            res = self.client.execute(query_str, {id_param: self.uid})
+            res = res["exportQueuedDataRows"]
+            if res["status"] == "COMPLETE":
+                return res["downloadUrl"]
+
+            timeout_seconds -= sleep_time
+            if timeout_seconds <= 0:
+                return None
+
+            logger.debug(
+                "Project '%s' queued data row export, waiting for server...",
+                self.uid)
+            time.sleep(sleep_time)
```

Comment:
nit: it's tempting to use `backoff.on_predicate` here (https://pypi.org/project/backoff/) not to reinvent polling logic, but might be not worth it - I'll leave it to you:)

Reply:
Good suggestion but I'm not going to add this. The user should be able to adjust the export time since a massive export could potentially take over 2 minutes.